### ISSUE TITLE
max object size

### ISSUE THREAD

**Issue by ordioma** [ultralytics/ultralytics#20054]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I encountered an issue during segmentation where my object occupies about 90% of the frame's width. When running inference with imgsz=640, everything works fine. However, when using imgsz=1280, the bounding box and mask don't cover the entire object. How can I handle this?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @ordioma, thank you for reaching out and for your interest in Ultralytics 🚀! We appreciate you sharing your question. An Ultralytics engineer will review your issue and provide assistance soon.

In the meantime, here are some steps that might help:

- If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us better understand and debug the issue you're experiencing.

- If this is a segmentation-related ❓ Question, please include as much detail as possible about your use case. This can include:
  - Dataset image examples
  - Training or inference logs
  - Your current model settings
  - Steps you've already taken to troubleshoot

We also recommend visiting the [Docs](https://docs.ultralytics.com/) for guidance. You can find usage examples for [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/), as well as tips for achieving the [Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

## Upgrade

Ensure you are using the latest version of `ultralytics` to verify the issue still persists. Update your package with:

```bash
pip install -U ultralytics
```

Check that your environment meets the required dependencies, including [Python>=3.8](https://www.python.org/) and [PyTorch>=1.8](https://pytorch.org/get-started/locally/).

## Environments

For a seamless experience, you can run YOLO in any of the following verified environments:
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Community

Join the Ultralytics community to connect with other users and share insights:
- [Discord](https://discord.com/invite/ultralytics) 🎧 for real-time chat
- [Discourse](https://community.ultralytics.com/) for in-depth discussions
- [Subreddit](https://reddit.com/r/Ultralytics) for community threads

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. These tests verify the correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across macOS, Windows, and Ubuntu.

We look forward to assisting you further! 😊

**Comment by Y-T-G** [comment]
You can try P6 model.
https://github.com/ultralytics/ultralytics/issues/16895#issuecomment-2413016050

But I am not sure why you need large imgsz if the object is already large.

**Comment by ordioma** [comment]
> You can try P6 model. [#16895 (comment)](https://github.com/ultralytics/ultralytics/issues/16895#issuecomment-2413016050)
> 
> But I am not sure why you need large imgsz if the object is already large.

Cause i need to segment a canvas, which can have tiny deffects on it.

**Comment by glenn-jocher** [comment]
Thanks for clarifying. Using a higher `imgsz` like 1280 makes sense for detecting tiny defects on large objects.

As suggested previously, P6 models are specifically designed for larger input resolutions (e.g., 1280x1280) compared to standard P5 models (typically 640x640). Using a P6 model variant (like `yolo11x-seg-p6.yaml` if available, or adapting a P6 config for training) might yield better segmentation results for large objects at `imgsz=1280` because its architecture is optimized for that scale.

**Comment by ordioma** [comment]
Where can i get pretrained yolo8m-seg-p6.pt? Pretrained model should train and perform better.

**Comment by glenn-jocher** [comment]
Currently, Ultralytics doesn't provide a pre-trained YOLOv8m-seg-p6 model in our official repository. The only pre-trained P6 model we currently offer is YOLOv8x-pose-p6.pt for pose estimation.

You have two options:
1. Use our latest YOLO11x-seg model which performs better on large objects at higher resolutions
2. Train your own P6 segmentation model by starting with YOLOv8m-seg.pt and a P6 configuration YAML

If you decide to train your own, you can modify the P6 configuration from one of our existing P6 model YAMLs in the repository and initialize training with the YOLOv8m-seg weights.
